detector_test.go

```diff
@@ -163,6 +164,39 @@ func testDetector(t *testing.T, when spec.G, it spec.S) {
 			}
 		})
 
+		it("should fail with specific error if any bp detect fails in an unexpected way", func() {
+			mkappfile("100", "detect-status")
+			mkappfile("0", "detect-status-A-v1")
+			mkappfile("-1", "detect-status-B-v1")
+			_, _, err := lifecycle.BuildpackOrder{
+				{Group: []lifecycle.Buildpack{
+					{ID: "A", Version: "v1", Optional: false},
+					{ID: "B", Version: "v1", Optional: false},
+				}},
+			}.Detect(config)
+			if err != lifecycle.ErrFailWithErrors {
+				t.Fatalf("Unexpected error:\n%s\n", err)
+			}
+
+			if runtime.GOOS == "windows" {
+				if s := allLogs(logHandler); !strings.HasSuffix(s,
+					"======== Results ========\n"+
+						"pass: A@v1\n"+
+						"err:  B@v1 (4294967295)\n",
```

Comment:
Is this a problem with the windows test buildpack? Shouldn't it select it's exit code from `detect-status-B-v1` and match the linux case?

Reply:
Pretty sure this was because I was returning `-1`, which invalid. I should have caught it turning to `255` on mac, but I think the positive error code should be defined behavior and now match.